### PR TITLE
feat(tickets): add ticket transfer doctype

### LINKED ISSUE #400: [FEATURE] Automate Ticket Transfers
### Feature description

We need to automate ticket transfer process. Right now we need a dedicated person who looks into transfer queries and caters to them individually, it is very _inefficient_

Proposed solution: 

I am going with the assumption that we let the people do the monetary transactions with each other without FOSS United having any part in it. The organization's role will only be to change the ticket details

- Let's have a portal where people can transfer tickets
- People can enter their ticket ID and email + email of the person they want the ticket to be transferred
- Validation of Ticket ID + email. Once validated, email will be sent to the original ticket holder who will approve the request from email.
- Upon approval, the ticket details will be updated and a new email will be sent to the person receiving the ticket.

### PR DESCRIPTION
**Pull request by harshtandiya**
## What type of PR is this? (check all applicable)
- [x] 🍕Feature

## Description

Automate the ticket transfer process. More details at #400 

## Related Issues & Docs
closes #400 

## Screenshots
image
image

### CODE REVIEW

File: dashboard/src/pages/TicketTransfer.vue [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+          <FormControl
+            v-model="reciever.reciever_email"
+            label="Reciever Email &ast;"
+            type="email"
+            size="sm"
+            placeholder="john@fossunited.org"
+          />
+          <FormControl
+            v-model="reciever.reciever_name"
+            label="Reciever Name &ast;"
```
Comment (RiyaMathew-11) [inline dashboard/src/pages/TicketTransfer.vue]: Typo in this - Reciever -> Receiver 

Needs to be changed in other places too.

**Review by RiyaMathew-11 — CHANGES_REQUESTED** [review]

File: dashboard/src/pages/TicketTransfer.vue [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+const ticketValidateError = ref('')
+const ticketErrors = ref('')
+
+const ticketId = ref('')
+const reciever = reactive({
+  reciever_email: '',
+  reciever_name: '',
+  designation: '',
+  organization: '',
```
Comment (RiyaMathew-11) [inline dashboard/src/pages/TicketTransfer.vue:155]: meta comment - if the purchased ticket has a tshirt alongside - then that would need to be transferred too right? 

How would the case be for different tshirt sizes for transferer and receiver ?

**Comment by RiyaMathew-11** [comment]
Can you also illustrate the case of how transfer approval and rejection status comes in ? 

When does a person see that message? - I am assuming that there is some time gap that would take in the approval process - how is a person able to track his/her ticket transfer status?

Is it in the format of a portal page that the transferer can access ?

File: dashboard/src/pages/TicketTransfer.vue (hunk 2 above)
Comment (harshtandiya) [reply dashboard/src/pages/TicketTransfer.vue:155]: Yes the tshirt will be transferred too.

I can add something for different tshirt size, but will that be okay? I hope there are no problems from logistic side of things

File: dashboard/src/pages/TicketTransfer.vue (hunk 1 above)
Comment (harshtandiya) [reply dashboard/src/pages/TicketTransfer.vue]: Outdated commit. Changed in later ones

**Comment by harshtandiya** [comment]
@RiyaMathew-11 

Notification to the owner on ticket transfer, where they can approve or reject.

On Approval -> redirect to website which runs script and changes the status of ticket transfer to completed

Another Notification -> Email to the new owner + old owner that ticket was transferred, also email to new owner with a ticket.pdf

File: dashboard/src/pages/TicketTransfer.vue (hunk 2 above)
Comment (RiyaMathew-11) [reply dashboard/src/pages/TicketTransfer.vue:155]: Well if incorrect tshirt sizes - I think that would also not work? Let's check with Ansh/Vishal on this too.

**Comment by RiyaMathew-11** [comment]
On second thoughts, I feel a little additional step here - an additional email notification for approval. 

The original owner transfers the ticket (only the original owner would have ticket ID - that's like a security key too)- after the process is initiated, why would he/she have to approve/reject it again via CTA from an email?  

I think it would be more easier if - once the `initiate transfer` button is checked, a floating window appears to confirm the transfer again - with Approve/Reject CTA and close the process there ? It reduces one set of emails. 

cc: @rahulporuri @NagariaHussain would like your thoughts too.

**Comment by harshtandiya** [comment]
@RiyaMathew-11 I think it's always better to be on the safer side. We should keep the email approval workflow.

File: dashboard/src/pages/TicketTransfer.vue (hunk 1 above)
Comment (RiyaMathew-11) [reply dashboard/src/pages/TicketTransfer.vue]: Okay - please update the screenshots in PR description too - it shows the typo

**Comment by RiyaMathew-11** [comment]
Okay, I still do feel, it's an add-on workflow when it could be simpler, but we could go with what works best.

**Pull request merged**